app/components/sidebar_item/sidebar_item.rb

```diff
@@ -6,11 +6,24 @@ def initialize(active: false, **)
       super
 
       @active = active
-      @styles << :active if active
     end
 
     private
 
-    attr_reader :active
+    def styles
+      return super << :active if active?
+
+      super
+    end
+
+    def attrs
+      return super.merge(data: { two_column_layout_target: 'activeItem' }) if active?
```

Comment:
You told me about the `::target` pseudo-class. https://developer.mozilla.org/en-US/docs/Web/CSS/:target